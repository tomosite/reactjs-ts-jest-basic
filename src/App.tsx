import { Container, Header } from 'semantic-ui-react'
import './App.css';
import { NaviItem } from './components';


function App() {

  return (
    <>
      <Container>
        <NaviItem />
      </Container>
    </>
  );
}

export default App;
