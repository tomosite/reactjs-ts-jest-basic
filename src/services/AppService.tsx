export const clickLink = (linkId: string) => {
  console.log(linkId);
};

export const AppService = {
  clickLink
};
