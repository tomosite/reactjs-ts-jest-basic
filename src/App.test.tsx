import { fireEvent, render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import { AppService } from './services/AppService';
import App from './App';

describe('App', () => {
  // jest.mock('./services/AppService');

  test('click the navigation link item', () => {
    const clickItemMock = jest.spyOn(AppService, 'clickLink');
    render(<App />);
    const link_elem = screen.getByText(/link item 1/i);
    expect(link_elem).toMatchSnapshot();
    fireEvent.click(link_elem);
    expect(clickItemMock).toHaveBeenCalledWith('2');
  });
});
