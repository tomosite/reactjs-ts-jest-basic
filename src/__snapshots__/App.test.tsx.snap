// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App click the navigation link item 1`] = `
<div
  class="item"
  role="listitem"
>
  item dummy 1
</div>
`;
